tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp

```diff
@@ -0,0 +1,319 @@
+  const size_t number_of_points = get<0>(x).size();
+  auto solution_at_six_times =
+      make_array<6>(Variables<valencia_tags>(number_of_points));
+
+  using solution_tags =
+      tmpl::list<hydro::Tags::RestMassDensity<DataVector>,
+                 hydro::Tags::SpecificInternalEnergy<DataVector>,
+                 hydro::Tags::SpatialVelocity<DataVector, 3, Frame::Inertial>,
+                 hydro::Tags::MagneticField<DataVector, 3, Frame::Inertial>,
+                 hydro::Tags::DivergenceCleaningField<DataVector>,
+                 hydro::Tags::LorentzFactor<DataVector>,
+                 hydro::Tags::Pressure<DataVector>,
+                 hydro::Tags::SpecificEnthalpy<DataVector>,
+                 gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>,
+                 gr::Tags::SqrtDetSpatialMetric<DataVector>>;
+
+  for (size_t i = 0; i < 6; ++i) {
+    const auto vars =
+        solution.variables(x, gsl::at(six_times, i), solution_tags{});
+
+    const auto& rest_mass_density =
+        get<hydro::Tags::RestMassDensity<DataVector>>(vars);
+    const auto& specific_internal_energy =
+        get<hydro::Tags::SpecificInternalEnergy<DataVector>>(vars);
+    const auto& spatial_velocity =
+        get<hydro::Tags::SpatialVelocity<DataVector, 3>>(vars);
+    const auto& magnetic_field =
+        get<hydro::Tags::MagneticField<DataVector, 3>>(vars);
+    const auto& divergence_cleaning_field =
+        get<hydro::Tags::DivergenceCleaningField<DataVector>>(vars);
+    const auto& lorentz_factor =
+        get<hydro::Tags::LorentzFactor<DataVector>>(vars);
+    const auto& pressure = get<hydro::Tags::Pressure<DataVector>>(vars);
+    const auto& specific_enthalpy =
+        get<hydro::Tags::SpecificEnthalpy<DataVector>>(vars);
+    const auto& spatial_metric =
+        get<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>>(vars);
+    const auto& sqrt_det_spatial_metric =
+        get<gr::Tags::SqrtDetSpatialMetric<DataVector>>(vars);
+
+    grmhd::ValenciaDivClean::ConservativeFromPrimitive::apply(
+        make_not_null(&get<grmhd::ValenciaDivClean::Tags::TildeD>(
+            gsl::at(solution_at_six_times, i))),
+        make_not_null(&get<grmhd::ValenciaDivClean::Tags::TildeTau>(
+            gsl::at(solution_at_six_times, i))),
+        make_not_null(&get<grmhd::ValenciaDivClean::Tags::TildeS<>>(
+            gsl::at(solution_at_six_times, i))),
+        make_not_null(&get<grmhd::ValenciaDivClean::Tags::TildeB<>>(
+            gsl::at(solution_at_six_times, i))),
+        make_not_null(&get<grmhd::ValenciaDivClean::Tags::TildePhi>(
+            gsl::at(solution_at_six_times, i))),
+        rest_mass_density, specific_internal_energy, specific_enthalpy,
+        pressure, spatial_velocity, lorentz_factor, magnetic_field,
+        sqrt_det_spatial_metric, spatial_metric, divergence_cleaning_field);
+  }
+
+  return (-1.0 / (60.0 * delta_time)) * solution_at_six_times[0] +
```

Comment:
Where are these numbers coming from?

Reply:
this is a sixth-order central FD stencil (I'll add a comment)